src/App/Pages/Accounts/LockPageViewModel.cs

```diff
@@ -173,6 +180,14 @@ public async Task InitAsync()
 
             BiometricEnabled = await _vaultTimeoutService.IsBiometricLockSetAsync() && await _cryptoService.HasEncryptedUserKeyAsync();
 
+            var decryptOptions = await _stateService.GetAccountDecryptionOptions();
+            if (await _stateService.IsAuthenticatedAsync() && decryptOptions?.TrustedDeviceOption != null
+                && (!decryptOptions?.HasMasterPassword ?? false && !(BiometricEnabled || PinEnabled)))
```

Comment:
⛏️ When you are checking `decryptOptions?.TrustedDeviceOption != null` it means that after that `decryptOptions` won't be `null` so no need to check it again afterwards so `!decryptOptions?.HasMasterPassword ?? false` can be just `!decryptOptions.HasMasterPassword`

Also no need for the parenthesis in 
`&& (!decryptOptions.HasMasterPassword && !(BiometricEnabled || PinEnabled)))`.

Finally, using De Morgan's law you can invert the biometric/pin enabled to avoid the parenthesis as well:
```suggestion
            if (await _stateService.IsAuthenticatedAsync() 
                && decryptOptions?.TrustedDeviceOption != null
                && !decryptOptions.HasMasterPassword 
                && !BiometricEnabled 
                && !PinEnabled)
```
♻️  It's a big if, maybe extracting this into a method would clarify the reading.

Reply:
Refactored the condition but didn't extracted it to a method because couldn't get a good name for it, it's just what it is, authed with tde, no password, has no biometrics nor pin 😅